rasterio/merge.py

```diff
@@ -300,8 +300,7 @@ def nullcontext(obj):
                 if cmath.isfinite(nodataval):
                     info = np.finfo(dt)
                     inrange = info.min <= nodataval <= info.max
-                    # check casting between float and complex
-                    inrange = inrange & np.can_cast(nodataval, dt)
+                    inrange = inrange
```

Comment:
Is this change intended?

Reply:
@groutr that `can_cast()` usage is invalid with NumPy 2.0.0. Removing it didn't break any test.

Reply:
I'll need to look at the tests again. It is supposed to check the case of a complex nodata value and a float datatype. We can't represent the complex nodata value as float.